### ISSUE TITLE
`earthly --push github.com/remote/project+target` does not push to registry

### ISSUE THREAD

**Issue by alexcb** [earthly/earthly#2234]
VERSION 0.6 should push images to a docker registry when called on a remote repo; however a user reported it is not working.

**Comment by rrjjvv** [comment]
I believe I've stumbled upon this as well.  I have some other non-trivial things happening in my setup as well, so it took me a while to think that it might be as simple as remote vs. local targets.  Despite some of the potential factors that could make me 'special' (private/self-hosted git repo, private/self-hosted registry, and remote buildkit as the more obvious ones), this report is matching up with my experience.

You specifically call out `VERSION 0.6`, suggesting that it (or the feature flags that it implies) might be responsible.  Is that the case?

This borders on hearsay, but in my earlier days of earthly, I recall running into an issue where running earthly in "artifact mode" wouldn't actually generate artifacts when using a remote target.  I didn't file a report at that time, chalking it up to my relative inexperience and that it wasn't a hindrance.  I haven't tried reproducing it lately; I'm only mentioning it in case it's a clue (like a more general issue with the push/save phase of remote target builds).

If there's any info I can provide to help, do let me know.  I can try to create a minimal reproducer if it'll help, with the caveat that you'd have to rely on console output (as we don't use publicly accessible git repos or docker registries).

**Comment by rrjjvv** [comment]
I can also add that what matters is where the `SAVE IMAGE --push` is defined, not the target that's invoked.  In other words, doing
```Earthfile
foo:
    BUILD +github.com/remote/project+target
```
with
```
earthly --push +foo
```
is not a sufficient workaround.  If you do a local clone of `github.com/remote/project` and then `earthly --push +target`, things work as expected.

**Comment by rrjjvv** [comment]
I was trying to diagnose an unrelated issue, but in the process, came across the code that dictates what's saved/pushed.  There's a couple places that are candidate culprits (to my untrained eyes):

https://github.com/earthly/earthly/blob/d5d239540b11e1b024c5c6349ae98ce2d82836ae/builder/builder.go#L294
https://github.com/earthly/earthly/blob/d5d239540b11e1b024c5c6349ae98ce2d82836ae/builder/builder.go#L590
https://github.com/earthly/earthly/blob/d5d239540b11e1b024c5c6349ae98ce2d82836ae/builder/builder.go#L294

It's hard to follow what's going on, but assuming those are related to this issue, it seems like this limitation was very deliberate and not a bug.  It could make sense if it was around "trust" in some way?

**Comment by NilsKintscher** [comment]
I stumpled upon this issue using a remote repo on bitbucket. I'll check if  this is fixed in VERSION 0.7.
